src/it/scala/com/databricks/spark/redshift/RedshiftIntegrationSuite.scala

```diff
@@ -386,6 +386,93 @@ class RedshiftIntegrationSuite extends IntegrationSuiteBase {
     }
   }
 
+  test("configuring compression on columns") {
+    val tableName = s"configuring_compression_on_columns_$randomSuffix"
+    try {
+      val metadata = new MetadataBuilder().putString("encoding", "LZO").build()
+      val schema = StructType(
+        StructField("x", StringType, metadata = metadata) :: Nil)
+      sqlContext.createDataFrame(sc.parallelize(Seq(Row("a" * 512))), schema).write
```

Comment:
This test seems to fail with a "java.sql.SQLException: Error (code 1204) while loading data into Redshift: "String length exceeds DDL length"" exception. My hunch is that compression doesn't affect data type constraints, such as maximum field length, and thus I think that you'll have to either reduce the string size or increase the column's string length.